c10/core/AllocatorConfig.cpp

```diff
@@ -0,0 +1,442 @@
+    TORCH_CHECK(
+        (config[i] == "True" || config[i] == "False"),
+        "Expected a single True/False argument for expandable_segments");
+    use_expandable_segments_ = (config[i] == "True");
+  } else {
+    TORCH_CHECK(false, "Error, expecting expandable_segments value");
+  }
+  return i;
+}
+
+size_t AllocatorConfig::parseReleaseLockOnDeviceMalloc(
+    const std::vector<std::string>& config,
+    size_t i) {
+  consumeToken(config, ++i, ':');
+  if (++i < config.size()) {
+    TORCH_CHECK(
+        (config[i] == "True" || config[i] == "False"),
+        "Expected a single True/False argument for release_lock_on_device_malloc, release_lock_on_cudamalloc or release_lock_on_hipmalloc");
+    use_release_lock_on_device_malloc_ = (config[i] == "True");
+  } else {
+    TORCH_CHECK(
+        false,
+        "Error, expecting release_lock_on_device_malloc, release_lock_on_cudamalloc or release_lock_on_hipmalloc value");
+  }
+  return i;
+}
+
+size_t AllocatorConfig::parsePinnedUseDeviceHostRegister(
+    const std::vector<std::string>& config,
+    size_t i) {
+  consumeToken(config, ++i, ':');
+
+  if (++i < config.size()) {
+    TORCH_CHECK(
+        (config[i] == "True" || config[i] == "False"),
+        "Expected a single True/False argument for pinned_used_device_host_register, pinned_used_cuda_host_register or pinned_used_hip_host_register");
+    pinned_use_device_host_register_ = (config[i] == "True");
+  } else {
+    TORCH_CHECK(
+        false,
+        "Error, expecting pinned_used_device_host_register, pinned_used_cuda_host_register or pinned_used_hip_host_register value");
+  }
+  return i;
+}
+
+size_t AllocatorConfig::parsePinnedNumRegisterThreads(
+    const std::vector<std::string>& config,
+    size_t i) {
+  consumeToken(config, ++i, ':');
+  if (++i < config.size()) {
+    size_t val_env = stoi(config[i]);
+    TORCH_CHECK(
+        llvm::isPowerOf2_64(val_env),
+        "Number of register threads has to be power of 2");
+    auto max_threads = pinned_max_register_threads();
+    TORCH_CHECK(
+        val_env <= max_threads,
+        "Number of register threads should be less than or equal to ",
+        max_threads);
+    pinned_num_register_threads_ = val_env;
+  } else {
+    TORCH_CHECK(false, "Error, expecting pinned_num_register_threads value");
+  }
+  return i;
+}
+
+size_t AllocatorConfig::parsePinnedUseBackgroundThreads(
+    const std::vector<std::string>& config,
+    size_t i) {
+  consumeToken(config, ++i, ':');
+  if (++i < config.size()) {
+    TORCH_CHECK(
+        (config[i] == "True" || config[i] == "False"),
+        "Expected a single True/False argument for pinned_use_background_threads");
+    pinned_use_background_threads_ = (config[i] == "True");
+  } else {
+    TORCH_CHECK(false, "Error, expecting pinned_use_background_threads value");
+  }
+  return i;
+}
+
+void AllocatorConfig::parseArgs(const char* env) {
+  // The following option will be reset to its default value if not explicitly
```

Comment:
Here is a bit weird. The original code ensure that these options will be reset if no explicitly set each time. I don't why, but this PR intends to re-org instead of changing the logic.